doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1613 @@
+
+#### 2.2.4.4 Bridge port backup nexthop 
+Backup nexthop for Bridge port is a useful feature to avoid traffic loss during local ES link down scenarios.
+
+https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.7.5&id=29cfb2aaa4425a608651a05b9b875bc445394443
+
+The above patch was added in kernel v6.6. 
+
+
+<a id="225-MAC-learning-aging-and-move-handling"></a>
+### 2.2.5 MAC learning, aging, and move handling
+
+The below steps explain the MAC learning and ageing scenario in the EVPN MH network. Please refer to diagram used in sec. 1.2.1.4 Proxy Advertisement of Type-2 routes.
+1. MAC is learnt on Vtep-1:
+	- (a) MAC is learnt in hardware on interface PortChannel1 and STATE_FDB_TABLE entry is populated (type=dynamic).
+  - (b) Fdbsyncd installs MAC address into the kernel with state=NUD_REACHABLE, dev=PortChannel1.
+	- (c) FRR advertises MAC in Type-2 update with Proxy = 0.
+
+2. Type-2 update is received on Vtep-4:
+	- (a) FRR installs MAC in kernel with dev=PortChannel1, state=NUD_NOARP
+	- (b) Fdbsyncd listens to the above update from kernel and installs VXLAN_FDB_TABLE with type=controlPlane, ifname=PortChannel1, and state=remote. Note: A new FDB type is defined for synchronized entry. The type can be dataplane, controlplane or both.
+	- (c) Fdborch installs the MAC in hardware with mesh-bit set to avoid ageing.
+	- (d) FRR advertises Type-2 route with Proxy=1.
+
+3. Type-2 proxy update is received on Vtep-1:
+    - (a) Type-2 update from Vtep-4 is processed and FRR converts the FDB entry in kernel from NUD_REACHABLE to NUD_NOARP with NFEA_ACTIVITY_NOTIFY set.
+    - (b) Fdbsyncd receives the state change event from kernel and populates VXLAN_FDB_TABLE with ifname=PortChannel1, type=dynamic+controlPlane.
+    - (c) Fdborch observes that STATE_FDB_TABLE entry is already present. It updates the bitmap in the FDB cache as Local + Remote. The entry in the hardware is not updated and remains as dynamic.
+
+4. MAC ages out on Vtep-1:
+    - (a) L2 table update event from SAI is processed in Fdborch.
+    - (b) Fdborch updates STATE_FBD_TABLE entry with type=controlPlane, and resets Local flag in FDB cache. It observes that Remote flag is set in FDB cache, so it updates MAC into the hardware with mesh bit set.
+    - (c) Fdbsyncd receives STATE_FDB_TABLE updates event. It sees type=controlPlane. It removes FDB entry from the kernel (or resets activity bit, if possible.)
+    - (d) FRR receives FDB notification with inactive_bit set. BGP withdraws Type-2 update.
+
+5. Type-2 withdrawal is received on Vtep-4:
+    - (a) FRR updates the kernel FDB entry with IN_TIMER flag and starts hold-timer.
+    - (c) Fdbsyncd receives notification from kernel with IN_TIMER flag set, and it replaces the VXLAN_FDB_TABLE entry with ageing=enabled, type=none.
+    - (d) Fdborch removes the mesh bit from the FDB entry in HW.
+    - (e) MAC learn event is received from SAI if the traffic hits after mesh bit is removed.
```

Comment:
After step 5.d, the FDB entry programmed in the hardware should be as below: MAC=H2,Dest=PO1, SAI_FDB_ENTRY_TYPE_DYNAMIC.
This is how any MAC entry learnt locally would have been programmed. The standard SAI behavior is to NOT generate learn/move events when it receives packets with SMAC as H2 and ingress port as PO1; since these do NOT need any control plane handling and sending these events would waste CPU cycles-- imagine an elephant flow with SMAC as H2 being received on PO1. 

Given the above, at step 5.e, why should SAI generate a learn/move event when it receives a packet with SMAC as H2 and ingress port as PO1? If we need a special behavior for a specific scenario, then we need new SAI attributes to indicate this behavior.